plonky2/src/gates/gate.rs

```diff
@@ -171,10 +178,18 @@ impl<F: RichField + Extendable<D>, const D: usize> Debug for GateRef<F, D> {
     }
 }
 
+/// Map between gate parameters and available slots.
+#[derive(Clone, Debug)]
+pub struct CurrentSlot<F: RichField + Extendable<D>, const D: usize> {
+    pub current_slot: HashMap<Vec<F>, (usize, usize)>,
+}
+
 /// A gate along with any constants used to configure it.
+#[derive(Clone)]
 pub struct GateInstance<F: RichField + Extendable<D>, const D: usize> {
     pub gate_ref: GateRef<F, D>,
     pub constants: Vec<F>,
+    pub params: Vec<F>,
```

Comment:
I know params are used in `find_slot`, but it seems like this field and the argument to `add_gate` aren't used, unless I missed something.

Reply:
Oh good catch! I used it in a previous version, but you're right that it's unnecessary now. I fixed it 👍